samples/javascript_nodejs/51.teams-messaging-extensions-action/README.md

```diff
@@ -1,94 +1,70 @@
-- Open the ```appsettings.json``` file
-- Paste your botID value into the ```MicrosoftAppId``` field 
-- Put the password into the ```MicrosoftAppPassword``` field
-- Save the file
-- Open the ```manifest.json```
-- Replace your botID everywhere you see the place holder string ```<YOUR-BOT-ID>```
-
-
-- Run the bot from a terminal
-
-  ```bash
-  npm install
-  ```
-  
-  ```bash
-  npm start
-  ```
-
-### Teams - App Studio
-- Launch Microsoft Teams
-- In the bar at the top of Teams search for and select ```App Studio``` 
-- Click the ```Manifest editor``` tab
-- Click ```Import an existing app```
-- Navigate to and select the `manifest.json` file from the previous step
-- Click on the `Action Messaging Extension` card
-- Click ```Test and distribute``` on the left hand side
-- Click the ```Install``` button
-- Click ```Add``` button
+1) Run ngrok - point to port 3978
+
+    ```bash
+    ngrok http -host-header=rewrite 3978
+    ```
+1) Create [Bot Framework registration](https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/create-a-bot-for-teams#register-your-web-service-with-the-bot-framework), using the current https URL you were given by running ngrok.
+
+1) Update the `.env` configuration for the bot to use the app id and app password from the Bot Framework registration. (Note the app password is referred to as the client secret in the azure portal and you can always create a new client secret anytime.)
+
+1) *This step is specific to Teams.* **Edit** the `manifest.json` contained in the  `teamsAppManifest` folder to replace your app if from Bot Framework everywhere you see the place holder string `<<YOUR-MICROSOFT-BOT-ID>>`.
+**Zip** up the contents of the `teamsAppManifest` folder to create a `manifest.zip`. **Upload** the `manifest.zip` to Teams (in the Apps view click "Upload a custom app")
+
+1) Run your bot at the command line:
+
+    ```bash
+    npm start
+    ```
 
 ### Interacting with the Messaging Extension
 
-1. Selecting the **Create Card** command from the Compose Box command list. The parameters dialog will be displayed and can be submitted to initiate the card creation within the Messaging Extension code. 
+> Note this `manifest.json` specified that the bot will be called from both the `compose` and `message` areas of Teams. Please refer to Teams documentation for more details.
+
+1) Selecting the **Create Card** command from the Compose Box command list. The parameters dialog will be displayed and can be submitted to initiate the card creation within the Messaging Extension code. 
 
 or
 
-2. Selecting the **Share Message** command from the Message command list.  
+2) Selecting the **Share Message** command from the Message command list.  
+
+## Deploy the bot to Azure
+
+To learn more about deploying a bot to Azure, see [Deploy your bot to Azure](https://aka.ms/azuredeployment) for a complete list of deployment instructions.
+
+## Further reading
```

Comment:
Further reading: https://docs.microsoft.com/en-us/microsoftteams/platform/messaging-extensions/how-to/action-commands/define-action-command